src/fu-main.c

```diff
@@ -206,16 +207,26 @@ fu_main_engine_percentage_changed_cb (FuEngine *engine,
 				       g_variant_new_uint32 (percentage));
 }
 
-static gboolean
-fu_main_get_device_flags_for_sender (FuMainPrivate *priv, const char *sender,
-				     FwupdDeviceFlags *flags, GError **error)
+static FuEngineRequest *
+fu_main_create_request (FuMainPrivate *priv, const gchar *sender, GError **error)
 {
-	uid_t calling_uid;
+	FwupdFeatureFlags *feature_flags;
+	FwupdDeviceFlags device_flags = FWUPD_DEVICE_FLAG_NONE;
+	uid_t calling_uid = 0;
+	g_autoptr(FuEngineRequest) request = fu_engine_request_new ();
 	g_autoptr(GVariant) value = NULL;
 
-	g_return_val_if_fail (sender != NULL, FALSE);
-	g_return_val_if_fail (flags != NULL, FALSE);
+	g_return_val_if_fail (sender != NULL, NULL);
+
+	/* did the client set the list of supported feature */
+	feature_flags = g_hash_table_lookup (priv->sender_features, sender);
+	if (feature_flags == NULL) {
+		fu_engine_request_set_feature_flags (request, FWUPD_FEATURE_FLAG_NONE);
```

Comment:
If you do take my other suggestion to set the defaults to none, this can probably be cleaned up that feature_flags set is only called when you have new features to add.

Reply:
Done.